### PR TITLE
Facebook OAuth and data sync

### PR DESCRIPTION
**Pull request by leVirve**
same to #53 
But PR from my `dev` branch for quick update

### CODE REVIEW

File: member/models.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from data_center.models import Course
+
+
+class Member(models.Model):
+    user = models.OneToOneField(User)
+
+    # Facebook field
+    uuid = models.CharField('User ID', max_length=100)
+    email = models.EmailField('Email')
+    access_token = models.TextField('access token')
```
Comment (afq984) [inline member/models.py]: tokens will expire according to facebook
https://developers.facebook.com/docs/facebook-login/access-tokens

File: search/views.py [hunk 2]
```diff
@@ -105,13 +107,35 @@ def syllabus(request, id):
...
+def course_manipulation(request, id):
+    """ Use this `course_manipulation` function to record the course """
+    if request.user.is_authenticated():
+        course = get_object_or_404(Course, id=id)
+        request_type = request.GET.get('type', 'GET')
+
+        if request_type == 'PUT':
+            request.user.member.courses.add(course)
+        elif request_type == 'DELETE':
+            request.user.member.courses.remove(course)
```
Comment (afq984) [inline search/views.py:116]: django supports `request.method == 'PUT'` but not `request.PUT`

actually, if you do this

``` bash
curl server_address -X any_method_name
```

you will get

``` python
request.method == 'ANY_METHOD_NAME'
```

in django

File: member/models.py (hunk 1 above)
Comment (leVirve) [inline member/models.py]: Yes, but it'll update each time as the user log in. (Refer to `member/views.py` Line.12 )
But we may not need this to get more information from facebook ? maybe I' ll remove this field.

File: search/views.py (hunk 2 above)
Comment (leVirve) [inline search/views.py:116]: Ok, I'll try it. thanks
But, exculding `GET`, csrf token needs to be included to achieve the same thing.

File: member/models.py (hunk 1 above)
Comment (afq984) [inline member/models.py]: I just think storing tokens is not a good idea haha

**Pull request closed without merging**